### DISCUSSION TITLE
How to code URLs for static files in an additional directory

### DISCUSSION THREAD

**Discussion by belfasthenry** [mojolicious/mojo#1764] [Q&A] [answered]
In my system I have an existing directory `/d/u/PhotoIndex` which contains my archive of photographs, organised into a tree of directories.  I have a Mojo application which reads the directories and sub-directories and presents web pages containing thumbnails of the photographs, with links to display them high-res.

My problem is that I'm generating URLs into that file tree which look right but don't display images.  That directory is in the static paths:
```perl script/pas.pl eval -V 'app->static->paths'
[
  "/home/henry/gitr/pas/public",
  "/d/u/PhotoIndex"
]
```
(`/home/henry/gitr/pas/public` is the public directory in the Mojo application tree).

The parts of the HTML which are supposed to display thumbnails of the images are like this:
```
<a href="/view/d/u/PhotoIndex/2016/10/2016_10_05%20Northumberland/IMG_0113.JPG" target="_blank">
      <img alt="IMG_0113.JPG" class="dirlist" src="/d/u/PhotoIndex/2016/10/2016_10_05%20Northumberland/IMG_0113.JPG">
```
.. but the images referenced by the "src" attribute are not being rendered.

(BTW "view" is a route I've defined to display the full-resolution image in a separate window))

My understanding of the process for serving static files is that Mojo searches the paths in order for the URL supplied; but I've done something wrong, because the file referred to above (IMG_0113.JPG) most certainly exists:
```
ls "/d/u/PhotoIndex/2016/10/2016_10_05 Northumberland/IMG_0113.JPG"
'/d/u/PhotoIndex/2016/10/2016_10_05 Northumberland/IMG_0113.JPG'
```
I've tried coding a absolute URL but relative to `/d/u/PhotoIndex` thus `src="/2016/10 ...etc` but that doesn't do what I want either.

**Comment by Grinnz** [comment, marked as answer]
If your static path is `/d/u/PhotoIndex` then the URL to `/d/u/PhotoIndex/2016/10/2016_10_05 Northumberland/IMG_0113.JPG` would be `/2016/10/2016_10_05 Northumberland/IMG_0113.JPG`.

**Reply by Grinnz** [reply]
Beyond that, check the log and the file/directory permissions.

**Comment by belfasthenry** [comment]
D'you know what ... I thought I'd tried that.  But plainly I hadn't, because it works fine.  Thank you.
